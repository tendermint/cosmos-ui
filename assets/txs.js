module.exports = [
  {
    hash: "not a real hash",
    time: 1042 + 1000, // set by Voyager
    height: "3436",
    tx: {
      type: "8EFE47F0625DE8",
      value: {
        msg: [
          {
            type: "cosmos-sdk/Send",
            value: {
              inputs: [
                {
                  coins: [
                    {
                      denom: "jbcoins",
                      amount: 1234
                    }
                  ],
                  address: "tb1da6xsetjg9jxgun9wdesexv05j"
                }
              ],
              outputs: [
                {
                  coins: [
                    {
                      denom: "jbcoins",
                      amount: 1234
                    }
                  ],
                  address: "tb1d4u5zerywfjhxuc9nudvw"
                }
              ]
            }
          }
        ],
        fee: {
          amount: [
            {
              denom: "",
              amount: 0
            }
          ],
          gas: 500000
        },
        gas: 500000,
        signatures: [
          {
            pub_key: {
              type: "AC26791624DE60",
              value: "70df5mUA3bFTJheAI6umXvABNuUU+q/3nOPS6lmT3tU="
            },
            signature: {
              type: "6BF5903DA1DB28",
              value:
                "oODzZz2t1u678Jhy2zFnFbBu7S2IP1zjOIoyVDOvGkQ6WkiiRF3zxEJbNAjHGdDjXf7/NQpMOtpt8FC1x6e6DA=="
            },
            account_number: 28,
            sequence: 0
          }
        ]
      },
      result: {
        gasUsed: "3324",
        tags: [
          {
            key: "c2VuZGVy",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          },
          {
            key: "cmVjaXBpZW50",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          }
        ],
        fee: {}
      }
    }
  },
  {
    hash: "not a real hash2",
    time: 1042 + 1010, // set by Voyager
    height: "3438",
    tx: {
      type: "8EFE47F0625DE8",

      value: {
        msg: [
          {
            type: "cosmos-sdk/Send",
            value: {
              inputs: [
                {
                  coins: [
                    {
                      denom: "jbcoins",
                      amount: 1234
                    }
                  ],
                  address: "tb1d4u5zerywfjhxuc9nudvw"
                }
              ],
              outputs: [
                {
                  coins: [
                    {
                      denom: "jbcoins",
                      amount: 1234
                    }
                  ],
                  address: "tb1da6xsetjg9jxgun9wdesexv05j"
                }
              ]
            }
          }
        ],
        fee: {
          amount: [
            {
              denom: "",
              amount: 0
            }
          ],
          gas: 500000
        },
        gas: 500000,
        signatures: [
          {
            pub_key: {
              type: "AC26791624DE60",
              value: "70df5mUA3bFTJheAI6umXvABNuUU+q/3nOPS6lmT3tU="
            },
            signature: {
              type: "6BF5903DA1DB28",
              value:
                "oODzZz2t1u678Jhy2zFnFbBu7S2IP1zjOIoyVDOvGkQ6WkiiRF3zxEJbNAjHGdDjXf7/NQpMOtpt8FC1x6e6DA=="
            },
            account_number: 28,
            sequence: 0
          }
        ]
      },
      result: {
        gasUsed: "3324",
        tags: [
          {
            key: "c2VuZGVy",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          },
          {
            key: "cmVjaXBpZW50",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          }
        ],
        fee: {}
      }
    }
  },
  {
    hash: "not a real hash3",

    time: 1042 + 100, // set by Voyager
    height: "466",
    tx: {
      type: "8EFE47F0625DE8",
      value: {
        msg: [
          {
            type: "cosmos-sdk/Send",
            value: {
              inputs: [
                {
                  coins: [
                    {
                      denom: "jbcoins",
                      amount: 1234
                    },
                    {
                      denom: "fabocoins",
                      amount: 1
                    },
                    {
                      denom: "mattcoins",
                      amount: 42
                    }
                  ],
                  address: "tb1da6xsetjg9jxgun9wdesexv05j"
                }
              ],
              outputs: [
                {
                  coins: [
                    {
                      denom: "jbcoins",
                      amount: 1234
                    },
                    {
                      denom: "fabocoins",
                      amount: 1
                    },
                    {
                      denom: "mattcoins",
                      amount: 42
                    }
                  ],
                  address: "tb1da6xsetjg9jxgun9wdesexv05j"
                }
              ]
            }
          }
        ],
        fee: {
          amount: [
            {
              denom: "",
              amount: 0
            }
          ],
          gas: 500000
        },
        gas: 500000,
        signatures: [
          {
            pub_key: {
              type: "AC26791624DE60",
              value: "70df5mUA3bFTJheAI6umXvABNuUU+q/3nOPS6lmT3tU="
            },
            signature: {
              type: "6BF5903DA1DB28",
              value:
                "oODzZz2t1u678Jhy2zFnFbBu7S2IP1zjOIoyVDOvGkQ6WkiiRF3zxEJbNAjHGdDjXf7/NQpMOtpt8FC1x6e6DA=="
            },
            account_number: 28,
            sequence: 0
          }
        ]
      },
      result: {
        gasUsed: "3324",
        tags: [
          {
            key: "c2VuZGVy",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          },
          {
            key: "cmVjaXBpZW50",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          }
        ],
        fee: {}
      }
    }
  },
  {
    hash: "not a real hash4",

    time: 1042 + 150, // set by Voyager
    height: "566",
    tx: {
      type: "8EFE47F0625DE8",
      value: {
        msg: [
          {
            type: "cosmos-sdk/MsgDelegate",
            value: {
              validator_addr:
                "cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw",
              delegation: {
                amount: 4200,
                denom: "steak"
              }
            }
          }
        ],
        fee: {
          amount: [
            {
              denom: "",
              amount: 0
            }
          ],
          gas: 500000
        },
        gas: 500000,
        signatures: [
          {
            pub_key: {
              type: "AC26791624DE60",
              value: "70df5mUA3bFTJheAI6umXvABNuUU+q/3nOPS6lmT3tU="
            },
            signature: {
              type: "6BF5903DA1DB28",
              value:
                "oODzZz2t1u678Jhy2zFnFbBu7S2IP1zjOIoyVDOvGkQ6WkiiRF3zxEJbNAjHGdDjXf7/NQpMOtpt8FC1x6e6DA=="
            },
            account_number: 28,
            sequence: 0
          }
        ]
      },
      result: {
        gasUsed: "3324",
        tags: [
          {
            key: "c2VuZGVy",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          },
          {
            key: "cmVjaXBpZW50",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          }
        ],
        fee: {}
      }
    }
  },
  {
    hash: "not a real hash5",

    time: 1042 + 150, // set by Voyager
    height: "566",
    tx: {
      type: "8EFE47F0625DE8",
      value: {
        msg: [
          {
            type: "cosmos-sdk/BeginUnbonding",
            value: {
              validator_addr:
                "cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw",
              shares: 100
            }
          }
        ],
        fee: {
          amount: [
            {
              denom: "",
              amount: 0
            }
          ],
          gas: 500000
        },
        gas: 500000,
        signatures: [
          {
            pub_key: {
              type: "AC26791624DE60",
              value: "70df5mUA3bFTJheAI6umXvABNuUU+q/3nOPS6lmT3tU="
            },
            signature: {
              type: "6BF5903DA1DB28",
              value:
                "oODzZz2t1u678Jhy2zFnFbBu7S2IP1zjOIoyVDOvGkQ6WkiiRF3zxEJbNAjHGdDjXf7/NQpMOtpt8FC1x6e6DA=="
            },
            account_number: 28,
            sequence: 0
          }
        ]
      },
      result: {
        gasUsed: "3324",
        tags: [
          {
            key: "c2VuZGVy",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          },
          {
            key: "cmVjaXBpZW50",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          }
        ],
        fee: {}
      }
    }
  }
]
