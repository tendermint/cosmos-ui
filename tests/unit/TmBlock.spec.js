import { mount } from "@vue/test-utils"
import htmlBeautify from "html-beautify"
import { TmBlock } from "@/index"
import txs from "../../assets/txs.js"
describe("TmBlock.vue", () => {
  let wrapper
  beforeEach(async () => {
    wrapper = mount(TmBlock, {
      propsData: {
        txs,
        currentUser: "tb1da6xsetjg9jxgun9wdesexv05j",
        validators: [
          {
            owner: "cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw",
            moniker: "cool validator"
          }
        ],
        loading: false,
        blockHeaderTime: "January 5th, 1988",
        blockMeta: {
          block_id: {
            hash: "asdf"
          }
        },
        block: {
          header: {
            chain_id: "1",
            time: "2",
            num_txs: "3",
            last_commit_hash: "4",
            validators_hash: "5",
            app_hash: "6",
            last_block_id: {
              hash: "7",
              parts: {
                total: "8",
                hash: "9"
              }
            }
          },
          last_commit: {
            precommits: [
              {
                validator_address: "10",
                validator_index: "11",
                round: "12",
                signature: {
                  type: "13",
                  data: "14"
                }
              }
            ]
          }
        }
      }
    })
  })

  it("has the expected html structure", () => {
    expect(htmlBeautify(wrapper.html())).toMatchSnapshot()
  })
})
