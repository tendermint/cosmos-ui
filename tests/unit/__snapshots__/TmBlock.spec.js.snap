// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmBlock.vue has the expected html structure 1`] = `
<div>
  <section class="tm-part">
    <div class="tm-part-container">
      <!---->
      <main class="tm-part-main">
        <div class="tm-li">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Block Hash
              </div>
              <div class="tm-li-dd">
                asdf
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </section>
  <section class="tm-part">
    <div class="tm-part-container">
      <header class="tm-part-header">
        <div class="tm-part-header-container">
          <div class="tm-part-title h5">
            Header
          </div>
          <menu class="tm-part-menu"></menu>
        </div>
      </header>
      <main class="tm-part-main">
        <div class="tm-li">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Chain ID
              </div>
              <div class="tm-li-dd">
                1
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Time
              </div>
              <div class="tm-li-dd">
                January 5th, 1988
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Transactions
              </div>
              <div class="tm-li-dd">
                3
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Last Commit Hash
              </div>
              <div class="tm-li-dd">
                4
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Validators Hash
              </div>
              <div class="tm-li-dd">
                5
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                App Hash
              </div>
              <div class="tm-li-dd">
                6
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </section>
  <section class="tm-part">
    <div class="tm-part-container">
      <header class="tm-part-header">
        <div class="tm-part-header-container">
          <div class="tm-part-title h5">
            Last Block
          </div>
          <menu class="tm-part-menu"></menu>
        </div>
      </header>
      <main class="tm-part-main">
        <div class="tm-li">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Hash
              </div>
              <div class="tm-li-dd">
                7
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Parts Total
              </div>
              <div class="tm-li-dd">
                8
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Parts Hash
              </div>
              <div class="tm-li-dd">
                9
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </section>
  <section class="tm-part">
    <div class="tm-part-container">
      <header class="tm-part-header">
        <div class="tm-part-header-container">
          <div class="tm-part-title h5">
            Precommit
          </div>
          <menu class="tm-part-menu"></menu>
        </div>
      </header>
      <main class="tm-part-main">
        <div class="tm-li">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Address
              </div>
              <div class="tm-li-dd">
                10
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Index
              </div>
              <div class="tm-li-dd">
                11
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Round
              </div>
              <div class="tm-li-dd">
                12
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Sig (13)
              </div>
              <div class="tm-li-dd">
                14
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </section>
  <section class="tm-part">
    <div class="tm-part-container">
      <header class="tm-part-header">
        <div class="tm-part-header-container">
          <div class="tm-part-title h5">
            Transactions
          </div>
          <menu class="tm-part-menu"></menu>
        </div>
      </header>
      <main class="tm-part-main">
        <div class="tm-li-tx">
          <div class="tm-li-tx__icon">
            <img src="../../assets/cosmos-logo.png" style="border-color: #25293f;">
          </div>
          <div class="tm-li-tx__content">
            <p class="tm-li-tx__content__caption">
              <div>
                Sent&nbsp;
                <b>1234</b><span>&nbsp;JBCOINS</span>
              </div>
            </p>
            <div class="tm-li-tx__content__information">
              <div class="tm-li-tx__content__information__details">
                <span>To tb1d4u5zerywfjhxuc9nudvw</span>
              </div>
              <div class="tm-li-tx__content__information__block">
                <a href="https://explorecosmos.network/blocks/3436">Block #3436&nbsp;</a>@ 2:03 am
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li-tx">
          <div class="tm-li-tx__icon">
            <img src="../../assets/cosmos-logo.png" style="border-color: #215c2f;">
          </div>
          <div class="tm-li-tx__content">
            <p class="tm-li-tx__content__caption">
              <div>
                Received&nbsp;
                <b>1234</b><span>&nbsp;JBCOINS</span>
              </div>
            </p>
            <div class="tm-li-tx__content__information">
              <div class="tm-li-tx__content__information__details">
                <span>From tb1d4u5zerywfjhxuc9nudvw</span>
              </div>
              <div class="tm-li-tx__content__information__block">
                <a href="https://explorecosmos.network/blocks/3438">Block #3438&nbsp;</a>@ 2:03 am
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li-tx">
          <div class="tm-li-tx__icon">
            <img src="../../assets/cosmos-logo.png" style="border-color: #25293f;">
          </div>
          <div class="tm-li-tx__content">
            <p class="tm-li-tx__content__caption">
              <div>
                Sent&nbsp;
                <b>1234</b><span>&nbsp;JBCOINS</span>
              </div>
            </p>
            <div class="tm-li-tx__content__information">
              <div class="tm-li-tx__content__information__details">
                <span>To yourself!</span>
              </div>
              <div class="tm-li-tx__content__information__block">
                <a href="https://explorecosmos.network/blocks/466">Block #466&nbsp;</a>@ 2:03 am
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li-tx">
          <div class="tm-li-tx__icon">
            <img src="../../assets/cosmos-logo.png" style="border-color: #512584;">
          </div>
          <div class="tm-li-tx__content">
            <p class="tm-li-tx__content__caption">
              <div>
                Delegated&nbsp;
                <b>4200</b><span>&nbsp;steak</span>
              </div>
            </p>
            <div class="tm-li-tx__content__information">
              <div class="tm-li-tx__content__information__details">
                <div>
                  To&nbsp;
                  <router-link href="/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw">
                    cool validator
                  </router-link>
                </div>
              </div>
              <div class="tm-li-tx__content__information__block">
                <a href="https://explorecosmos.network/blocks/566">Block #566&nbsp;</a>@ 2:03 am
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li-tx">
          <div class="tm-li-tx__icon">
            <img src="../../assets/cosmos-logo.png" style="border-color: #9d5c19;">
          </div>
          <div class="tm-li-tx__content">
            <p class="tm-li-tx__content__caption">
              <div>
                Unbonded&nbsp;
                <b>100</b><span>&nbsp;Steak</span>
              </div>
            </p>
            <div class="tm-li-tx__content__information">
              <div class="tm-li-tx__content__information__details">
                <div>
                  From&nbsp;
                  <router-link to="/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw">
                    cool validator
                  </router-link>
                </div>
              </div>
              <div class="tm-li-tx__content__information__block">
                <a href="https://explorecosmos.network/blocks/566">Block #566&nbsp;</a>@ 2:03 am
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </section>
</div>
`;
